access/grpc/convert.go

```diff
@@ -542,3 +542,107 @@ func messageToTransactionResult(m *access.TransactionResultResponse, options []j
 		CollectionID:  flow.BytesToID(m.GetCollectionId()),
 	}, nil
 }
+
+func messageToBlockExecutionData(
+	m *entities.BlockExecutionData,
+) (*flow.ExecutionData, error) {
+	if m == nil {
+		return nil, errEmptyMessage
+	}
+
+	chunks := make([]*flow.ChunkExecutionData, len(m.ChunkExecutionData))
+	for i, chunk := range m.GetChunkExecutionData() {
+		convertedChunk, err := messageToChunkExecutionData(chunk)
+		if err != nil {
+			return nil, err
+		}
+		chunks[i] = convertedChunk
+	}
+
+	return &flow.ExecutionData{
+		BlockID:            messageToIdentifier(m.GetBlockId()),
+		ChunkExecutionData: chunks,
+	}, nil
+}
+
+func messageToChunkExecutionData(
+	m *entities.ChunkExecutionData,
+) (*flow.ChunkExecutionData, error) {
+
+	transactions, err := messageToExecutionDataCollection(m.GetCollection())
+	if err != nil {
+		return nil, err
+	}
+
+	var trieUpdate *flow.TrieUpdate
+	if m.GetTrieUpdate() != nil {
+		trieUpdate, err = messageToTrieUpdate(m.GetTrieUpdate())
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	events := make([]*flow.Event, len(m.GetEvents()))
+
+	for i, ev := range m.GetEvents() {
+		res, err := messageToEvent(ev, nil)
+		if err != nil {
+			return nil, err
+		}
+		events[i] = &res
+	}
+
+	return &flow.ChunkExecutionData{
```

Comment:
this was recently updated to include transaction results

https://github.com/onflow/flow-go/blob/master/engine/common/rpc/convert/execution_data.go#L139